Json2Fs.Core/Library.fs

```diff
@@ -114,17 +117,68 @@ let rec generateFSharpDTOsByJson (typename : string) (json : JsonElement) : Type
     
     { props = props; typename = capitalize typename } :: addedTypes
 
-let fsRecordsToString (types : TypeToGenerate list) =
-    let escape s =
-        match s with
-        | "abstract" | "type" -> $"``{s}``"
-        | _ -> s
-
-    str {
-        for { props = props; typename = typename } in types |> List.rev do
-            $"type {typename} = {{\n"
-            for pair in props do
-                let typename = pair.Value |> List.rev |> String.concat " "
-                $"    {escape pair.Key} : {typename}\n"
-            "}\n"
-    }
+let private zeroRange = Range.Zero
+let private zeroXml = PreXmlDoc.Empty
+let private mkIdent s = Ident(s, zeroRange)
+let private mkSynTypeLongIdent s = SynType.LongIdent(LongIdentWithDots([mkIdent s], []))
+
+let fsRecordsToString (types: TypeToGenerate list) =
+    let mdls =
+        types
+        |> List.map (fun { typename = name; props = props } ->
+            let recordFields =
+                props
+                |> Map.toList
+                |> List.map (fun (key, value) ->
+                    let typeName =
+                        match value with
+                        | [] -> failwith "unexpected"
+                        | [ single ] -> mkSynTypeLongIdent single
+                        | head :: tail ->
+                            SynType.App(
+                                mkSynTypeLongIdent head,
+                                Some zeroRange,
+                                tail |> List.map mkSynTypeLongIdent,
+                                [],
+                                Some zeroRange,
+                                true,
+                                zeroRange
+                            )
+
+                    SynField([], false, Some(mkIdent key), typeName, false, zeroXml, None, zeroRange))
+
+            SynModuleDecl.Types(
+                [ SynTypeDefn.SynTypeDefn(
+                      SynComponentInfo([], None, [], [ mkIdent name ], zeroXml, false, None, zeroRange),
+                      SynTypeDefnRepr.Simple(SynTypeDefnSimpleRepr.Record(None, recordFields, zeroRange), zeroRange),
+                      [],
+                      None,
+                      zeroRange
+                  ) ],
+                zeroRange
+            ))
+
+    let file =
+        ParsedInput.ImplFile(
+            ParsedImplFileInput(
+                "tmp.fsx",
+                true,
+                QualifiedNameOfFile(mkIdent "Tmp"),
+                [],
+                [],
+                [ SynModuleOrNamespace(
+                      [],
+                      false,
+                      SynModuleOrNamespaceKind.AnonModule,
+                      mdls,
+                      zeroXml,
+                      [],
+                      None,
+                      zeroRange
+                  ) ],
+                (true, true)
+            )
+        )
+
+    CodeFormatter.FormatASTAsync(file, "tmp.fsx", [], None, FormatConfig.FormatConfig.Default)
```

Comment:
Okay, so I decided to get into it!

Just one minor thing, I think it'd be nicer to have a bit different formatting:

```suggestion
    let config = { FormatConfig.FormatConfig.Default with RecordMultilineFormatter = FormatConfig.MultilineFormatterType.NumberOfItems }

    CodeFormatter.FormatASTAsync(file, "tmp.fsx", [], None, config)
```

Reply:
I'll get to this later this week.